app/models/cached_forecast.rb

```diff
@@ -4,6 +4,12 @@ class CachedForecast < ApplicationRecord
   serialize :data
 
   def self.stale?
+    if (latest_record = last)
```

Comment:
`last` is calling the method `CachedForecast.last`.

I use the ["assignment in condition"](https://docs.rubocop.org/rubocop/cops_lint.html#lintassignmentincondition) technique:

```rb
if latest_record = last
```

as I don't want to call `last` twice e.g.

```rb
if last
  ...
  return last.obtained_at < threshold
end
```

I don't want to call last twice because it's making a db call and even though it will be very fast, this `CachedForecast#stale?` method is called a lot!

Reply:
so `latest_record = last` is assigning the return value of `CachedForecast.last` to `latest_record`. The inner code is only evaluated if the return value is not `nil` -- i.e. if the database isn't empty

Reply:
it's not a typo in a comparison (`latest_record == last`) which is why our linter requires it to be wrapped in parens...

Reply:
aha! thanks, i'm with you now